masonry/src/widget/zstack.rs

```diff
@@ -0,0 +1,324 @@
+            (VerticalAlignment::Center, HorizontalAlignment::Trailing) => Self::Trailing,
+            (VerticalAlignment::Bottom, HorizontalAlignment::Leading) => Self::BottomLeading,
+            (VerticalAlignment::Bottom, HorizontalAlignment::Center) => Self::Bottom,
+            (VerticalAlignment::Bottom, HorizontalAlignment::Trailing) => Self::BottomTrailing,
+        }
+    }
+
+    pub fn vertical(self) -> VerticalAlignment {
+        match self {
+            Alignment::Center | Alignment::Leading | Alignment::Trailing => {
+                VerticalAlignment::Center
+            }
+            Alignment::Top | Alignment::TopLeading | Alignment::TopTrailing => {
+                VerticalAlignment::Top
+            }
+            Alignment::Bottom | Alignment::BottomLeading | Alignment::BottomTrailing => {
+                VerticalAlignment::Bottom
+            }
+        }
+    }
+
+    pub fn horizontal(self) -> HorizontalAlignment {
+        match self {
+            Alignment::Center | Alignment::Top | Alignment::Bottom => HorizontalAlignment::Center,
+            Alignment::Leading | Alignment::TopLeading | Alignment::BottomLeading => {
+                HorizontalAlignment::Leading
+            }
+            Alignment::Trailing | Alignment::TopTrailing | Alignment::BottomTrailing => {
+                HorizontalAlignment::Trailing
+            }
+        }
+    }
+}
+
+impl Default for Alignment {
+    fn default() -> Self {
+        Self::Center
+    }
+}
+
+impl Default for VerticalAlignment {
+    fn default() -> Self {
+        Self::Center
+    }
+}
+
+impl Default for HorizontalAlignment {
+    fn default() -> Self {
+        Self::Center
+    }
+}
+
+impl From<Alignment> for VerticalAlignment {
+    fn from(value: Alignment) -> Self {
+        value.vertical()
+    }
+}
+
+impl From<Alignment> for HorizontalAlignment {
+    fn from(value: Alignment) -> Self {
+        value.horizontal()
+    }
+}
+
+impl From<(VerticalAlignment, HorizontalAlignment)> for Alignment {
+    fn from((vertical, horizontal): (VerticalAlignment, HorizontalAlignment)) -> Self {
+        Alignment::new(vertical, horizontal)
+    }
+}
+
+impl From<VerticalAlignment> for Alignment {
+    fn from(vertical: VerticalAlignment) -> Self {
+        Alignment::new(vertical, HorizontalAlignment::Center)
+    }
+}
+
+impl From<HorizontalAlignment> for Alignment {
+    fn from(horizontal: HorizontalAlignment) -> Self {
+        Alignment::new(VerticalAlignment::Center, horizontal)
+    }
+}
+
+// --- MARK: IMPL ZSTACK ---
+impl ZStack {
+    pub fn new() -> Self {
+        ZStack {
+            children: Vec::new(),
+            alignment: Alignment::default(),
+        }
+    }
+
+    pub fn with_alignment(mut self, alignment: impl Into<Alignment>) -> Self {
+        self.alignment = alignment.into();
+        self
+    }
+
+    pub fn with_child(self, child: impl Widget) -> Self {
```

Comment:
This needs to document whether the child is at the front or the back.

Reply:
Ideally in the name already, something like `with_appended_child`... (Though TBF `flex` and `grid` has the same names for these methods, so that's likely a case for a follow-up PR).

Reply:
I've written a fair bit of documentation in 42473c8, is there anything obvious that I missed?

Reply:
A comment has been added